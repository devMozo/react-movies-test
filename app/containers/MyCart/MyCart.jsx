import React from 'react';
import { connect } from 'react-redux';
import './MyCart.scss';
import CartItem from '../../components/CartItem/CartItem';
import { removeItem } from '../../redux-app/actions/MyCartAction';

export class MyCart extends React.PureComponent {
    onRemoveItem = itemId => {
        const { removeItem } = this.props;
        removeItem(itemId);
    };

    render() {
        const { items } = this.props;

        return (
            <section className="MyCart">
                {items && items.length > 0 ? (
                    <ul className="MyCart__list">
                        {items.map(item => (
                            <CartItem item={item} onRemove={this.onRemoveItem} />
                        ))}
                    </ul>
                ) : (
                    <p className="MyCart__empty-message"> There isn't an item yet </p>
                )}
            </section>
        );
    }
}

const mapStateToProps = state => {
    return {
        ...state.myCart,
    };
};

const mapDispatchToProps = dispatch => {
    return {
        removeItem: itemId => {
            dispatch(removeItem(itemId));
        },
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(MyCart);
