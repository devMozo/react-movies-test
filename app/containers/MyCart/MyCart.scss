.MyCart {
    * {
        font-family: 'Arial';
    }

    .MyCart__empty-message {
        text-align: center;
        margin-top: 60px;
        color: #666;
    }
}
