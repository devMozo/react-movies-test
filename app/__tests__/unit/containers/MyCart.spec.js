import React from 'react';
import { shallow, mount } from 'enzyme';
import { MyCart } from '../../../containers/MyCart/MyCart';
import { BrowserRouter as Router } from 'react-router-dom';

const MOCK_DISCOVER_MOVIE = require('../../../__mocks__/discoverMovie.json');

describe('<MyCart />', () => {
    let PROPS;
    const mockedOnremove = jest.fn();

    beforeEach(() => {
        PROPS = {
            removeItem: mockedOnremove,
            items: MOCK_DISCOVER_MOVIE.results,
        };
    });

    it('should render correctly in "debug" mode', () => {
        const component = shallow(<MyCart {...PROPS} />);
        expect(component).toMatchSnapshot();
    });

    it('should render a full list of items', () => {
        const component = mount(
            <Router>
                <MyCart {...PROPS} />
            </Router>,
        );

        expect(component.find('.CartItem').length).toBe(MOCK_DISCOVER_MOVIE.results.length);
    });
});
