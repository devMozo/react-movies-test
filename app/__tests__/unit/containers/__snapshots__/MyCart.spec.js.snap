// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyCart /> should render correctly in "debug" mode 1`] = `
<section
  className="MyCart"
>
  <ul
    className="MyCart__list"
  >
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/phxiKFDvPeQj4AbkvJLmuZEieDU.jpg",
          "genre_ids": Array [
            878,
            28,
          ],
          "id": 320288,
          "original_language": "en",
          "original_title": "Dark Phoenix",
          "overview": "The X-Men face their most formidable and powerful foe when one of their own, Jean Grey, starts to spiral out of control. During a rescue mission in outer space, Jean is nearly killed when she's hit by a mysterious cosmic force. Once she returns home, this force not only makes her infinitely more powerful, but far more unstable. The X-Men must now band together to save her soul and battle aliens that want to use Grey's new abilities to rule the galaxy.",
          "popularity": 246.408,
          "poster_path": "/kZv92eTc0Gg3mKxqjjDAM73z9cy.jpg",
          "release_date": "2019-06-05",
          "title": "Dark Phoenix",
          "video": false,
          "vote_average": 6.3,
          "vote_count": 879,
        }
      }
      key="320288"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/2FYzxgLNuNVwncilzUeCGbOQzP7.jpg",
          "genre_ids": Array [
            28,
            35,
            878,
            12,
          ],
          "id": 479455,
          "original_language": "en",
          "original_title": "Men in Black: International",
          "overview": "The Men in Black have always protected the Earth from the scum of the universe. In this new adventure, they tackle their biggest, most global threat to date: a mole in the Men in Black organization.",
          "popularity": 201.83,
          "poster_path": "/dPrUPFcgLfNbmDL8V69vcrTyEfb.jpg",
          "release_date": "2019-06-12",
          "title": "Men in Black: International",
          "video": false,
          "vote_average": 6.1,
          "vote_count": 255,
        }
      }
      key="479455"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/w2PMyoyLU22YvrGK3smVM9fW1jj.jpg",
          "genre_ids": Array [
            28,
            12,
            878,
          ],
          "id": 299537,
          "original_language": "en",
          "original_title": "Captain Marvel",
          "overview": "The story follows Carol Danvers as she becomes one of the universe’s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races. Set in the 1990s, Captain Marvel is an all-new adventure from a previously unseen period in the history of the Marvel Cinematic Universe.",
          "popularity": 177.765,
          "poster_path": "/AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg",
          "release_date": "2019-03-06",
          "title": "Captain Marvel",
          "video": false,
          "vote_average": 7,
          "vote_count": 5851,
        }
      }
      key="299537"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/vVpEOvdxVBP2aV166j5Xlvb5Cdc.jpg",
          "genre_ids": Array [
            80,
            28,
            53,
          ],
          "id": 458156,
          "original_language": "en",
          "original_title": "John Wick: Chapter 3 – Parabellum",
          "overview": "Super-assassin John Wick returns with a $14 million price tag on his head and an army of bounty-hunting killers on his trail. After killing a member of the shadowy international assassin’s guild, the High Table, John Wick is excommunicado, but the world’s most ruthless hit men and women await his every turn.",
          "popularity": 129.572,
          "poster_path": "/ziEuG1essDuWuC5lpWUaw1uXY2O.jpg",
          "release_date": "2019-05-15",
          "title": "John Wick: Chapter 3 – Parabellum",
          "video": false,
          "vote_average": 7.2,
          "vote_count": 1192,
        }
      }
      key="458156"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/qeFO3u2IqbAeIT1Xgo6POqYbCaQ.jpg",
          "genre_ids": Array [
            35,
            80,
            9648,
            28,
          ],
          "id": 514999,
          "original_language": "en",
          "original_title": "Murder Mystery",
          "overview": "After attending a gathering on a billionaire's yacht during a European vacation, a New York cop and his wife become prime suspects when he's murdered.",
          "popularity": 113.438,
          "poster_path": "/bSMSO9xupd4R4vwTPqigHn2quLN.jpg",
          "release_date": "2019-06-14",
          "title": "Murder Mystery",
          "video": false,
          "vote_average": 6.5,
          "vote_count": 543,
        }
      }
      key="514999"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/7RyHsO4yDXtBv1zUU3mTpHeQ0d5.jpg",
          "genre_ids": Array [
            12,
            878,
            28,
          ],
          "id": 299534,
          "original_language": "en",
          "original_title": "Avengers: Endgame",
          "overview": "After the devastating events of Avengers: Infinity War, the universe is in ruins due to the efforts of the Mad Titan, Thanos. With the help of remaining allies, the Avengers must assemble once more in order to undo Thanos' actions and restore order to the universe once and for all, no matter what consequences may be in store.",
          "popularity": 107.373,
          "poster_path": "/or06FN3Dka5tukK1e9sl16pB3iy.jpg",
          "release_date": "2019-04-24",
          "title": "Avengers: Endgame",
          "video": false,
          "vote_average": 8.4,
          "vote_count": 7098,
        }
      }
      key="299534"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg",
          "genre_ids": Array [
            12,
            28,
            14,
          ],
          "id": 299536,
          "original_language": "en",
          "original_title": "Avengers: Infinity War",
          "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.",
          "popularity": 104.105,
          "poster_path": "/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
          "release_date": "2018-04-25",
          "title": "Avengers: Infinity War",
          "video": false,
          "vote_average": 8.3,
          "vote_count": 14087,
        }
      }
      key="299536"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/aQXTw3wIWuFMy0beXRiZ1xVKtcf.jpg",
          "genre_ids": Array [
            28,
            878,
            53,
            12,
          ],
          "id": 399579,
          "original_language": "en",
          "original_title": "Alita: Battle Angel",
          "overview": "When Alita awakens with no memory of who she is in a future world she does not recognize, she is taken in by Ido, a compassionate doctor who realizes that somewhere in this abandoned cyborg shell is the heart and soul of a young woman with an extraordinary past.",
          "popularity": 100.437,
          "poster_path": "/xRWht48C2V8XNfzvPehyClOvDni.jpg",
          "release_date": "2019-01-31",
          "title": "Alita: Battle Angel",
          "video": false,
          "vote_average": 6.7,
          "vote_count": 2112,
        }
      }
      key="399579"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/wf6VDSi4aFCZfuD8sX8JAKLfJ5m.jpg",
          "genre_ids": Array [
            16,
            28,
            18,
            9648,
            35,
          ],
          "id": 566555,
          "original_language": "ja",
          "original_title": "名探偵コナン 紺青の拳（フィスト）",
          "overview": "23rd movie in the \\"Detective Conan\\" franchise.",
          "popularity": 95.075,
          "poster_path": "/86Y6qM8zTn3PFVfCm9J98Ph7JEB.jpg",
          "release_date": "2019-04-12",
          "title": "Detective Conan: The Fist of Blue Sapphire",
          "video": false,
          "vote_average": 4.4,
          "vote_count": 15,
        }
      }
      key="566555"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/uovH5k4BAEPqXqxgwVrTtqH169g.jpg",
          "genre_ids": Array [
            878,
            28,
            53,
          ],
          "id": 373571,
          "original_language": "en",
          "original_title": "Godzilla: King of the Monsters",
          "overview": "The new story follows the heroic efforts of the crypto-zoological agency Monarch as its members face off against a battery of god-sized monsters, including the mighty Godzilla, who collides with Mothra, Rodan, and his ultimate nemesis, the three-headed King Ghidorah. When these ancient super-species—thought to be mere myths—rise again, they all vie for supremacy, leaving humanity’s very existence hanging in the balance.",
          "popularity": 91.357,
          "poster_path": "/pU3bnutJU91u3b4IeRPQTOP8jhV.jpg",
          "release_date": "2019-05-29",
          "title": "Godzilla: King of the Monsters",
          "video": false,
          "vote_average": 6.3,
          "vote_count": 623,
        }
      }
      key="373571"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/ygWKYTu7OnZKF9H5NsgjL9iURGV.jpg",
          "genre_ids": Array [
            28,
            53,
          ],
          "id": 480042,
          "original_language": "en",
          "original_title": "Escape Plan: The Extractors",
          "overview": "TBD: The daughter of a Hong Kong tech executive goes missing in what appears to be a routine ransom job. As Breslin and his crew delve deeper, they discover the culprit is the deranged son of one of their former foes, who also kidnapped Breslin's love and is holding her inside the massive prison known as Devil's Station.",
          "popularity": 87.599,
          "poster_path": "/Au7tg0IcN9SDTEi2tlJwxB9RLYt.jpg",
          "release_date": "2019-06-20",
          "title": "Escape Plan: The Extractors",
          "video": false,
          "vote_average": 4.6,
          "vote_count": 8,
        }
      }
      key="480042"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/2JS5guOh3hj3WgMhbMHPZDSU1HJ.jpg",
          "genre_ids": Array [
            28,
            80,
          ],
          "id": 486131,
          "original_language": "en",
          "original_title": "Shaft",
          "overview": "JJ, aka John Shaft Jr., may be a cyber security expert with a degree from MIT, but to uncover the truth behind his best friend’s untimely death, he needs an education only his dad can provide. Absent throughout JJ’s youth, the legendary locked-and-loaded John Shaft agrees to help his progeny navigate Harlem’s heroin-infested underbelly.",
          "popularity": 76.808,
          "poster_path": "/kfZqwGuvEBAysAbCsa0QLKoSYR.jpg",
          "release_date": "2019-06-14",
          "title": "Shaft",
          "video": false,
          "vote_average": 5,
          "vote_count": 46,
        }
      }
      key="486131"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/nDP33LmQwNsnPv29GQazz59HjJI.jpg",
          "genre_ids": Array [
            9648,
            10751,
            35,
            878,
            28,
            12,
          ],
          "id": 447404,
          "original_language": "en",
          "original_title": "Pokémon Detective Pikachu",
          "overview": "In a world where people collect pocket-size monsters (Pokémon) to do battle, a boy comes across an intelligent monster who seeks to be a detective.",
          "popularity": 72.943,
          "poster_path": "/wgQ7APnFpf1TuviKHXeEe3KnsTV.jpg",
          "release_date": "2019-05-03",
          "title": "Pokémon Detective Pikachu",
          "video": false,
          "vote_average": 7,
          "vote_count": 1417,
        }
      }
      key="447404"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/dihW2yTsvQlust7mSuAqJDtqW7k.jpg",
          "genre_ids": Array [
            28,
            12,
            878,
          ],
          "id": 429617,
          "original_language": "en",
          "original_title": "Spider-Man: Far from Home",
          "overview": "Peter Parker and his friends go on a summer trip to Europe. However, they will hardly be able to rest - Peter will have to agree to help Nick Fury uncover the mystery of creatures that cause natural disasters and destruction throughout the continent.",
          "popularity": 71.521,
          "poster_path": "/dzg8lHS78l5Xxpd2EM5H541Qn2s.jpg",
          "release_date": "2019-06-28",
          "title": "Spider-Man: Far from Home",
          "video": false,
          "vote_average": 0,
          "vote_count": 0,
        }
      }
      key="429617"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/mFb0ygcue4ITixDkdr7wm1Tdarx.jpg",
          "genre_ids": Array [
            28,
            53,
          ],
          "id": 245891,
          "original_language": "en",
          "original_title": "John Wick",
          "overview": "Ex-hitman John Wick comes out of retirement to track down the gangsters that took everything from him.",
          "popularity": 71.127,
          "poster_path": "/b9uYMMbm87IBFOq59pppvkkkgNg.jpg",
          "release_date": "2014-10-22",
          "title": "John Wick",
          "video": false,
          "vote_average": 7.2,
          "vote_count": 10173,
        }
      }
      key="245891"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/uUiId6cG32JSRI6RyBQSvQtLjz2.jpg",
          "genre_ids": Array [
            28,
            12,
            16,
            878,
            35,
          ],
          "id": 324857,
          "original_language": "en",
          "original_title": "Spider-Man: Into the Spider-Verse",
          "overview": "Miles Morales is juggling his life between being a high school student and being a spider-man. When Wilson \\"Kingpin\\" Fisk uses a super collider, others from across the Spider-Verse are transported to this dimension.",
          "popularity": 68.819,
          "poster_path": "/iiZZdoQBEYBv6id8su7ImL0oCbD.jpg",
          "release_date": "2018-12-06",
          "title": "Spider-Man: Into the Spider-Verse",
          "video": false,
          "vote_average": 8.4,
          "vote_count": 4151,
        }
      }
      key="324857"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/VuukZLgaCrho2Ar8Scl9HtV3yD.jpg",
          "genre_ids": Array [
            878,
            28,
          ],
          "id": 335983,
          "original_language": "en",
          "original_title": "Venom",
          "overview": "Investigative journalist Eddie Brock attempts a comeback following a scandal, but accidentally becomes the host of Venom, a violent, super powerful alien symbiote. Soon, he must rely on his newfound powers to protect the world from a shadowy organization looking for a symbiote of their own.",
          "popularity": 68.653,
          "poster_path": "/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg",
          "release_date": "2018-09-28",
          "title": "Venom",
          "video": false,
          "vote_average": 6.6,
          "vote_count": 6308,
        }
      }
      key="335983"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/bi4jh0Kt0uuZGsGJoUUfqmbrjQg.jpg",
          "genre_ids": Array [
            28,
            12,
            35,
            14,
          ],
          "id": 287947,
          "original_language": "en",
          "original_title": "Shazam!",
          "overview": "A boy is given the ability to become an adult superhero in times of need with a single magic word.",
          "popularity": 65.257,
          "poster_path": "/xnopI5Xtky18MPhK40cZAGAOVeV.jpg",
          "release_date": "2019-03-23",
          "title": "Shazam!",
          "video": false,
          "vote_average": 7.1,
          "vote_count": 1912,
        }
      }
      key="287947"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/axjFiijtxdfK3CD9dMwrdgLwblD.jpg",
          "genre_ids": Array [
            12,
            28,
            53,
          ],
          "id": 375588,
          "original_language": "en",
          "original_title": "Robin Hood",
          "overview": "A war-hardened Crusader and his Moorish commander mount an audacious revolt against the corrupt English crown.",
          "popularity": 57.149,
          "poster_path": "/AiRfixFcfTkNbn2A73qVJPlpkUo.jpg",
          "release_date": "2018-11-20",
          "title": "Robin Hood",
          "video": false,
          "vote_average": 5.8,
          "vote_count": 1218,
        }
      }
      key="375588"
      onRemove={[Function]}
    />
    <Memo(CartItem)
      item={
        Object {
          "adult": false,
          "backdrop_path": "/2zKusjJZd8b2Cdk2C5K9dGFNWiJ.jpg",
          "genre_ids": Array [
            16,
            28,
            14,
            12,
          ],
          "id": 390634,
          "original_language": "ja",
          "original_title": "劇場版 Fate／stay night [Heaven's Feel] II. lost butterfly",
          "overview": "Theatrical-release adaptation of the visual novel \\"Fate/stay night\\", following the third and final route. (Part 2 of a trilogy.)",
          "popularity": 55.159,
          "poster_path": "/4tS0iyKQBDFqVpVcH21MSJwXZdq.jpg",
          "release_date": "2019-01-12",
          "title": "Fate/Stay Night: Heaven’s Feel II. Lost Butterfly",
          "video": false,
          "vote_average": 5.3,
          "vote_count": 54,
        }
      }
      key="390634"
      onRemove={[Function]}
    />
  </ul>
</section>
`;
